person.rb

```diff
@@ -0,0 +1,22 @@
+class Person
+  def initialize(age, name = 'unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_reader :id
+
+  attr_accessor :name, :age
```

Comment:
- Optional. 
By ruby conventions and best practices, the `attr_reader` and `attr_accessor` are generally mentioned above the `initialize` method. I would strongly recommend you move them to the top of the class.